torch/testing/_internal/common_distributed.py

```diff
@@ -350,14 +350,24 @@ def _spawn_processes(self):
         self._start_processes(proc)
 
     @classmethod
-    def _run(cls, rank, test_name, file_name):
+    def _run(cls, rank, test_name, file_name, pipe):
         self = cls(test_name)
         self.rank = rank
         self.file_name = file_name
 
         # self.id() == e.g. '__main__.TestDistributed.test_get_rank'
         # We're retrieving a corresponding test and executing it.
-        getattr(self, test_name)()
+        try:
+            getattr(self, test_name)()
+        except Exception as e:
+            logging.error(
+                'Caught exception: \n{}exiting process with exit code: {}'
+                .format(traceback.format_exc(), MultiProcessTestCase.TEST_ERROR_EXIT_CODE)
+            )
+            # Send error to parent process.
+            pipe.send(traceback.format_exc())
+            pipe.close()
```

Comment:
Do we need to close the pipe when we don't send and there are no errors or is close only for telling the other process no more data will be sent?